### PR TITLE
Extract docs base

### PR DESCRIPTION
**Pull request by SvenDowideit**
This PR extracts the docs html theme and generation tooling so that it can be used from a Hub image for all docker projects.

the goal is to be able to run `make docs` in any of our repo's, and have that build that repo's documentation set. And then to use the same tooling to publish to docs.docker.com.

This PR removes the theme and extra tooling, and also demos incorporating the compose, machine and swarm docs _and_ adds the v1 image spec to the references menu (see the `docs/Dockerfile`)

The removed common parts are currently in https://github.com/SvenDowideit/docs (and would move to `docker/docs`), which is set up to build as `docs/base:latest`.

the Dockerfile brings the source needed together, and then each repo's mkdocs.yaml gets imported to make the final artifact.
- [x] Docker Machine is merged - you can run `make docs` in a `docker/machine` checkout to preview documentation changes.
- [x] Docker Compose PR has been made, waiting for merge to I can update the PR to use it
- [x] Docker Swarm PR has been made, waiting for merge to I can update the PR to use it

### CODE REVIEW

**Comment by SvenDowideit** [comment]
@tianon @crosbymichael @spf13 @tiborvass @jfrazelle I'm interested to know what you think.

**Comment by jfrazelle** [comment]
Sorry Gordon is freaking out, contain yourself gordon

**Comment by jfrazelle** [comment]
Ok it's GitHub it's firing this hook multiple times... I'll patch Gordon tomorrow to check for existing comments

**Comment by SvenDowideit** [comment]
atm, the results are at http://dowideit-docs.s3-website-ap-southeast-2.amazonaws.com/

so it works end to end like before, but adds non docs dir files and will allow other repo docs builds (demo PR for that to come in those repos)

**Comment by moxiegirl** [comment]
My understanding is that this will allow a sub project like libcontainer to build its own documentation.  Based on that understanding, I think the idea is moving in the right direction especially in light of the plans for 1.6.   I have questions tho and they may be newb questions:

* Who has requested this functionality? Are they waiting for it or are we giving people something we think is good for them.

* There are existing concerns with our existing docs build (for example, use of push -f) does this spread an existing problem around to other users 

*  We have problems with the existing documentation; For example, the lack of secondary submenus from the main and just this morning I discovered that the search returns **HIDDEN** results.  Are we spreading problems?

* @spf13 Initiated an email this morning regarding changing out our tool. I think that is a good idea given the tool is admittedly defficient and poorly adopted compared to similar.  So, are we doing work we will undo or redo within two months and do the benefits outweigh the cost?

* Who will be supporting the other new "users" of the theme --- I'm too new so I'm not sure if the subsystems are local to us or not.  Anyway, we are short staffed as it is so if we need to support other folks outside of Docker (or even within) is this a good idea to do this now?

Those are my immediate thoughts and based on my admittedly new and perhaps uniformed understanding of both the tools and the projects involved.

**Comment by fredlf** [comment]
As a near-term fix to some complexities and issues we have publishing docs from sub-projects, this looks fine to me. It does not add to our support load. Basically, it keeps things running while we discuss/decide on tooling changes, etc.

File: docs/Dockerfile [hunk 1]
```diff
@@ -1,86 +1,44 @@
 #
 # See the top level Makefile in https://github.com/docker/docker for usage.
 #
-FROM debian:jessie
+FROM docs/base:latest
```
Comment (tianon) [inline docs/Dockerfile]: Hah -- I thought this was some strange hard-coded assumption, but no, you actually have control of this namespace: https://registry.hub.docker.com/u/docs/base/

File: Makefile [hunk 2]
```diff
@@ -85,11 +84,7 @@ build: bundles
 
 docs-build:
 	git fetch https://github.com/docker/docker.git docs && git diff --name-status FETCH_HEAD...HEAD -- docs > docs/changed-files
-	cp ./VERSION docs/VERSION
-	echo "$(GIT_BRANCH)" > docs/GIT_BRANCH
-	echo "$(AWS_S3_BUCKET)" > docs/AWS_S3_BUCKET
-	echo "$(GITCOMMIT)" > docs/GITCOMMIT
-	docker build -t "$(DOCKER_DOCS_IMAGE)" docs
+	docker build -t "$(DOCKER_DOCS_IMAGE)" -f docs/Dockerfile .
```
Comment (tianon) [inline Makefile]: Using `-f` here essentially means that to canonically build our docs requires Docker version 1.5.0+.  I'm not sure how to feel about that, and would appreciate more input from other maintainers WRT this.

File: docs/Dockerfile (hunk 1 above)
Comment (SvenDowideit) [inline docs/Dockerfile]: :green_apple:

File: Makefile (hunk 2 above)
Comment (SvenDowideit) [inline Makefile]: agreed. 

I can remove this part if it is concerning, as it is only there to bring in the image spec - and thus isn't necessary for thursday.

LMK

**Comment by SvenDowideit** [comment]
@jfrazelle is @GordonTheTurtle's whinging due to a bug? or do my PR's have some magical whitespace that the checker code can't deal with?

**Comment by jfrazelle** [comment]
I have never seen him misbehave so much other than on your PRs, stop
breaking him ;) jk I'll look into it

**Comment by jfrazelle** [comment]
it's also only this PR lol

File: Makefile (hunk 2 above)
Comment (tiborvass) [inline Makefile]: If the alternative is easy, let's do the alternative.

**Comment by thaJeztah** [comment]
^^ this PR has two commits, would that make a difference?

File: Makefile (hunk 2 above)
Comment (SvenDowideit) [inline Makefile]: roger that

**Comment by jfrazelle** [comment]
why not make it a repo on the docker account

**Comment by SvenDowideit** [comment]
@jfrazelle because i don't have access - if someone makes a `docker/docs` repo and gives me and the others access, I'll push it there.

**Comment by jfrazelle** [comment]
ah ok just wondering ;)

**Comment by jfrazelle** [comment]
seems fine to me and if we do use another markdown static site builder we would just update it at the other repo. i mean -32,000 is pretty cool

**Comment by fredlf** [comment]
LGTM (thought I did above, but forgot to, apparently, sry)

**Comment by SvenDowideit** [comment]
Last 2 commits to allow us to manually place the orchestration docs in the menus, and to remove the versioning drop down, as it won't work for them.

**Comment by aluzzardi** [comment]
Hey @jfrazelle, is something blocking this PR or are we good to go? Thanks! :)

**Comment by jfrazelle** [comment]
lgtm

**Comment by tiborvass** [comment]
@SvenDowideit @aanand

```
docker run --rm -it  -e AWS_S3_BUCKET -e NOCACHE -p 8000:8000 "docker-docs:HEAD" mkdocs serve
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 4, in <module>
    __import__('pkg_resources').run_script('mkdocs==0.9', 'mkdocs')
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 534, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 1438, in run_script
    execfile(script_filename, namespace, namespace)
  File "/usr/local/lib/python2.7/dist-packages/mkdocs-0.9-py2.7.egg/EGG-INFO/scripts/mkdocs", line 46, in <module>
    main(cmd, args=sys.argv[2:], options=dict(opts))
  File "/usr/local/lib/python2.7/dist-packages/mkdocs-0.9-py2.7.egg/EGG-INFO/scripts/mkdocs", line 29, in main
    serve(config, options=options)
  File "/usr/local/lib/python2.7/dist-packages/mkdocs-0.9-py2.7.egg/mkdocs/serve.py", line 84, in serve
    build(config, live_server=True)
  File "/usr/local/lib/python2.7/dist-packages/mkdocs-0.9-py2.7.egg/mkdocs/build.py", line 225, in build
    build_pages(config)
  File "/usr/local/lib/python2.7/dist-packages/mkdocs-0.9-py2.7.egg/mkdocs/build.py", line 186, in build_pages
    html_content = post_process_html(html_content, site_navigation)
  File "/usr/local/lib/python2.7/dist-packages/mkdocs-0.9-py2.7.egg/mkdocs/build.py", line 70, in post_process_html
    html_content = re.sub(r'a href="([^"]*)"', PathToURL(nav), html_content)
  File "/usr/lib/python2.7/re.py", line 151, in sub
    return _compile(pattern, flags).sub(repl, string, count)
  File "/usr/local/lib/python2.7/dist-packages/mkdocs-0.9-py2.7.egg/mkdocs/build.py", line 34, in __call__
    assert False, msg % (source_file, target_file)
AssertionError: The page "compose/index.md" contained a hyperlink to "compose/django.md" which is not listed in the "pages" configuration.
```

**Comment by jfrazelle** [comment]
INFO[0002] Tag latest not found in repository docs/base 
Makefile:89: recipe for target 'docs-build' failed
make: *** [docs-build] Error 1

**Comment by SvenDowideit** [comment]
ah joy, changes while I was sleeping :)

**Comment by SvenDowideit** [comment]
@tiborvass PTAL - there were some new docs added in the fig markdown that needed adding to the menu control file in this PR

**Comment by tiborvass** [comment]
Docs are there, `make docs` works as expected.

**Comment by tiborvass** [comment]
@SvenDowideit http://beta-docs.docker.io/userguide/ is missing links to those 3 pages, unless we don't want to because /swarm is not at /userguide/swarm.

**Comment by SvenDowideit** [comment]
we do not. the swarm userguide was decided to be at /swarm/, and all other swarm docs under that dir.

**Comment by SvenDowideit** [comment]
I'm not sure i understand what you're seeing/saying - the userguide menu i see on that site contains the swarm/compose/machine docs - the URL is not under `/userguide` but that's because all the swarm docs are under the `/swarm/ URL

**Comment by tiborvass** [comment]
@SvenDowideit i was just saying that at the bottom of http://beta-docs.docker.io/userguide/ the swarm/compose/machine docs are not there, but "Working with Docker Hub" is there. Not sure if that's intended.

**Comment by SvenDowideit** [comment]
@tiborvass oooooh, k, i'll add that, one mo

**Comment by tiborvass** [comment]
LGTM

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**